### PR TITLE
Release 0.7.0

### PR DESCRIPTION
**Pull request by blootsvoets**
Changes since version 0.6.0:
- Use coroutines for a larger subset of functions
- Updated authentication handling to radar-auth 2.1.0
- Updated radar-commons to 1.1.1 including plugin handling
- More consistent Avro context handling while parsing

### CODE REVIEW

File: src/integrationTest/kotlin/org/radarbase/gateway/resource/KafkaTopicsTest.kt [hunk 1]
```diff
@@ -90,217 +90,246 @@ class KafkaTopicsTest {
...
+            bearer(gatewayContext.accessToken)
+        }.let { response ->
+            assertThat(response.status, equalTo(HttpStatusCode.OK))
         }
 
-        httpClient.call(Status.UNAUTHORIZED) {
-            url("$BASE_URI/topics")
+        httpClient.get("$BASE_URI/topics").let { response ->
+            assertThat(response.status, equalTo(HttpStatusCode.Unauthorized))
         }
```
Comment (Bdegraaf1234) [inline src/integrationTest/kotlin/org/radarbase/gateway/resource/KafkaTopicsTest.kt:166]: Does this need to happen twice?

File: src/integrationTest/kotlin/org/radarbase/gateway/resource/KafkaTopicsTest.kt [hunk 2]
```diff
@@ -90,217 +90,246 @@ class KafkaTopicsTest {
...
+            assertThat(response.status, equalTo(HttpStatusCode.UnprocessableEntity))
         }
 
-        httpClient.call(Status.BAD_REQUEST) {
-            url("$BASE_URI/topics/test")
-            post("".toRequestBody(KAFKA_JSON_TYPE))
-            addHeader("Authorization", BEARER + accessToken)
+        httpClient.post("$BASE_URI/topics/test") {
+            bearer(gatewayContext.accessToken)
+            setBody("{}")
```
Comment (Bdegraaf1234) [inline src/integrationTest/kotlin/org/radarbase/gateway/resource/KafkaTopicsTest.kt]: this is a repeat of the last test if i'm not mistaken

```suggestion
            setBody("")
```

File: src/integrationTest/kotlin/org/radarbase/gateway/resource/KafkaTopicsTest.kt [hunk 3]
```diff
@@ -90,217 +90,246 @@ class KafkaTopicsTest {
...
+            contentType(ContentType.Application.Json)
+        }.let { response ->
+            assertThat(response.status, equalTo(HttpStatusCode.UnprocessableEntity))
```
Comment (Bdegraaf1234) [inline src/integrationTest/kotlin/org/radarbase/gateway/resource/KafkaTopicsTest.kt]: see above
```suggestion
            assertThat(response.status, equalTo(HttpStatusCode.BadRequest))
```

**Comment by Bdegraaf1234** [comment]
Some small things in one of the testfiles, otherwise everything looks good :)

File: src/integrationTest/kotlin/org/radarbase/gateway/resource/KafkaTopicsTest.kt (hunk 1 above)
Comment (blootsvoets) [reply src/integrationTest/kotlin/org/radarbase/gateway/resource/KafkaTopicsTest.kt:166]: Fixed

File: src/integrationTest/kotlin/org/radarbase/gateway/resource/KafkaTopicsTest.kt (hunk 3 above)
Comment (blootsvoets) [reply src/integrationTest/kotlin/org/radarbase/gateway/resource/KafkaTopicsTest.kt]: Fixed

**Pull request merged**